lib/module-resolvers/file-resolver.js

```diff
@@ -24,6 +27,22 @@ class FileResolver {
 
         this._additional_search_paths = options.search_paths || search_paths();
 
+        // Also add the module root for all adapters.
+        this._adapter_roots = [];
+
+        for(let adapter of adapters.list()) {
+            // Skip built-in adapters or adapters for which the
+            // install path is not absolute (i.e. not actually available)
+            if (adapter.module !== '(builtin)' &&
+                path.isAbsolute(adapter.path)) {
+                let install_path = this._canonicalize(adapter.path);
+                this._adapter_roots.push({
+                    adapter: adapter.adapter,
+                    module_root: [path.dirname(install_path), ADAPTER_MODULE_SUBDIR].join(path.sep)
+                });
```

Comment:
Maybe add the path only when it exists and it is a directory? Otherwise we will unnecessarily check it on every module load.